ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeFileLoader.java

```diff
@@ -16,86 +17,119 @@ public class BehaviorTreeFileLoader<T extends BehaviorTreeNodeLayer<T, ?, ?, ?>>
 {
    private final BehaviorTreeState behaviorTreeState;
    private final BehaviorTreeNodeStateBuilder<T> nodeBuilder;
+   private final WorkspaceResourceDirectory treeFilesDirectory;
 
-   public BehaviorTreeFileLoader(BehaviorTreeState behaviorTreeState, BehaviorTreeNodeStateBuilder<T> nodeBuilder)
+   public BehaviorTreeFileLoader(BehaviorTreeState behaviorTreeState,
+                                 BehaviorTreeNodeStateBuilder<T> nodeBuilder,
+                                 WorkspaceResourceDirectory treeFilesDirectory)
    {
       this.behaviorTreeState = behaviorTreeState;
       this.nodeBuilder = nodeBuilder;
+      this.treeFilesDirectory = treeFilesDirectory;
    }
 
-   public T loadFromFile(WorkspaceResourceFile fileToLoad, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
+   public T loadFromFile(WorkspaceResourceFile file, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
+   {
+      return loadFromFile(file, null, null, topologyOperationQueue);
+   }
+
+   private T loadFromFile(WorkspaceResourceFile file, JsonNode jsonNode, T parentNode, BehaviorTreeTopologyOperationQueue topologyOperationQueue)
```

Comment:
I've restructured this to be a more well-formed recursive function. This allowed to reduce some code duplication and fix some bugs.